pandas/core/internals/blocks.py

```diff
@@ -781,12 +781,14 @@ def _replace_coerce(
             )
         else:
             if value is None:
-                # gh-45601, gh-45836
-                nb = self.astype(np.dtype(object), copy=False)
-                if nb is self and not inplace:
-                    nb = nb.copy()
-                putmask_inplace(nb.values, mask, value)
-                return [nb]
+                # gh-45601, gh-45836, gh-46634
+                if mask.any():
+                    nb = self.astype(np.dtype(object), copy=False)
+                    if nb is self and not inplace:
+                        nb = nb.copy()
+                    putmask_inplace(nb.values, mask, value)
+                    return [nb]
+                return [self]
```

Comment:
probably need self.copy() if not inplace?

Reply:
hmm, in #46404 (the cause of the latest regression), I reverted some of the code change in #44940 (the cause of the previous regression). In this PR I have reverted some more of the changes in #44940 that obviously (now!) should also have been included in the previous regression fix. I don't see the original code making a copy if not mask.any().

but yes, in #44940 the code path was changed to dispatch to Block.replace which does make a copy even if the replace is a no-op. So as a pure regression fix we could keep this as is, but also happy to do a copy for consistency with other replacements in 1.4.x

my biggest concern (need to check this though) is categorical since the special casing was also moved in #44940.

I will mark this a draft for now, (till we get closer to the next release) and see if we can get the issues on master sorted so that we can dispatch to Block.replace when the replacement value is `None`. (doing that might end up being difficult to backport though)

Reply:
I'm pretty actively trying to dis-engage, not gonna fuss about it either way.  i trust your judgment here.

Reply:
There is an issue raised for the categorical issue #46672, so can keep that out of this PR for now.

> I will mark this a draft for now, (till we get closer to the next release) and see if we can get the issues on master sorted so that we can dispatch to Block.replace when the replacement value is `None`. (doing that might end up being difficult to backport though)

I guess no harm in getting this bit sorted and then doing that as a separate PR. (maybe after #46672 is sorted).